Automattic/wp-calypso
# Replace CurrentPlanHeader with MyPlanCard on /plans/my-plan page

This PR is a first step towards adding products to `/plans/my-plan` page.

It replaces the usage of `<CurrentPlanHeader />` with a new component `<PurchasesList />`. It, in turn, uses a newly-introduced `<MyPlanCard />` design component.

The logic that was inside the legacy component is now broken down into multiple new methods in the `<PurchasesList />` component.

The next step will be to add support for Jetpack products in the same parent component.

**Before:**
![Screenshot 2019-11-15 at 18 13 05](https://user-images.githubusercontent.com/478735/68961887-a1aa2e00-07d3-11ea-83b7-d2338867962d.png)

**After:**
![Screenshot 2019-11-15 at 18 12 12](https://user-images.githubusercontent.com/478735/68961901-a8d13c00-07d3-11ea-8df6-ed78e45c6e53.png)

**Before:**
![Screenshot 2019-11-15 at 18 12 40](https://user-images.githubusercontent.com/478735/68961891-a373f180-07d3-11ea-9950-34a24668e8eb.png)

**After:**
![Screenshot 2019-11-15 at 18 11 45](https://user-images.githubusercontent.com/478735/68961907-ab339600-07d3-11ea-9749-bef6e055c2af.png)

#### Changes proposed in this Pull Request

* Replace `CurrentPlanHeader` with `PurchasesList` and `MyPlanCard` on `/plans/my-plan` page

#### Testing instructions



* Check out this branch
* Go to http://calypso.localhost:3000/plans/my-plan/
* Switch between all kinds of sites (free and paid plans, Jetpack sites free and paid, Pressable, sites that you're not an owner of etc.) and check if there are no regressions.

Fixes n/a
